### PR TITLE
Upgrade pandas pinnings to support `2.2.3`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Pandas released a newer version `2.2.3` with very minimal fixes but one that adds support for python-3.13 and numpy 2.1 compatibility. This PR updates pinnings in `cudf` to support `2.2.3`.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by jameslamb — APPROVED** [review]

**Review by vyasr — APPROVED** [review]

**Comment by vyasr** [comment]
@galipremsagar there seems to be a consistent failure of the CUDA 12.01 Python 3.11 job. I doubt the CUDA version has anything to do with it, but Python 3.11 seems like a much more likely culprit. Could you take a look?

**Comment by galipremsagar** [comment]
> @galipremsagar there seems to be a consistent failure of the CUDA 12.01 Python 3.11 job. I doubt the CUDA version has anything to do with it, but Python 3.11 seems like a much more likely culprit. Could you take a look?

Yup

**Review by wence- — APPROVED** [review]

File: ci/test_python_common.sh [hunk 1]
```diff
@@ -17,6 +17,9 @@ rapids-dependency-file-generator \
   --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION};dependencies=${RAPIDS_DEPENDENCIES}" \
     | tee "${ENV_YAML_DIR}/env.yaml"
 
+rapids-logger "env file"
+cat "${ENV_YAML_DIR}/env.yaml"
+
```
Comment (galipremsagar) [inline ci/test_python_common.sh:22]:
```suggestion
```

**Review by vyasr — APPROVED** [review]
This looks great, thanks for sorting out the arm solver issue!

**Pull request merged**